src/lib.rs

```diff
@@ -92,8 +136,10 @@ pub struct WriteIter<'b, 'a: 'b, T: 'a> {
 impl<T> Storage<T> {
     fn from_inner(inner: StorageInner<T>) -> Storage<T> {
         let pending = Pending {
-            add_ref: Vec::new(),
-            sub_ref: Vec::new(),
+            add_ref_strong: Vec::new(),
+            sub_ref_strong: Vec::new(),
+            add_ref_weak: Vec::new(),
```

Comment:
I wonder why do we even need to store the weak counters? They aren't holding anything alive anyway, so we might as well not bother.